components/layout_2020/table/layout.rs

```diff
@@ -120,9 +123,12 @@ impl CollapsedBorder {
         Self { style_color, width }
     }
 
-    fn from_style(style: &ComputedValues, writing_mode: WritingMode) -> LogicalSides<Self> {
-        let border_style_color = style.border_style_color(writing_mode);
-        let border_width = style.border_width(writing_mode);
+    fn from_layout_style(
+        layout_style: &dyn LayoutStyle,
```

Comment:
Instead of using `dyn`, I think it would be better to make `LayoutStyle` an enum and implement the methods in questions directly on the `enum`. Having a trait implementation like this makes it a bit harder to trace what code is actually running during a function call.

Reply:
The problem with this approach is that there is a bunch of private table logic that I can't access from `style_ext.rs`

Reply:
You could use the same two classes that you had defined before, but make the methods that were trait implementations before `pub(crate)` and do this:

```rust
enum LayoutStyle {
    Default(DefaultLayoutStyle),
    Table(TableLayoutStyle),
}
```

Reply:
Yeah, that works! But for now `LayoutStyle::Table` is still not using layout information so I guess it's fine as is.